fake_joint_driver/src/fake_joint_driver.cpp

```diff
@@ -25,103 +24,145 @@ FakeJointDriver::FakeJointDriver(void)
+            std::back_inserter(joint_names_));
   // Check the emptyness of joints
   if (joint_names_.size() == 0) {
     ROS_ERROR("No joints is specified. Please use include_joints parameters.");
     ros::shutdown();
   }
   // Resize members
   cmd_dis.resize(joint_names_.size());
+  cmd_dis_old.resize(joint_names_.size());
+  cmd_vel.resize(joint_names_.size(), 0.0);
   act_dis.resize(joint_names_.size());
   act_vel.resize(joint_names_.size());
   act_eff.resize(joint_names_.size());
 
   // Set start position
-  for (auto it=start_position_map.begin(); it!=start_position_map.end(); it++)
-  {
-    for (auto i=0; i<joint_names_.size(); i++)
-    {
-      if (joint_names_[i] == it->first)
-      {
+  for (auto it = start_position_map.begin(); it != start_position_map.end();
+       it++) {
+    for (auto i = 0; i < joint_names_.size(); i++) {
+      if (joint_names_[i] == it->first) {
         act_dis[i] = it->second;
         cmd_dis[i] = it->second;
+        cmd_dis_old[i] = cmd_dis[i];
       }
     }
   }
 
   // Create joint_state_interface and position_joint_interface
-  for (int i = 0; i< joint_names_.size(); i++)
-  {
+  for (int i = 0; i < joint_names_.size(); i++) {
     ROS_DEBUG_STREAM("joint[" << i << "]:" << joint_names_[i]);
     // Connect and register the joint_state_interface
-    hardware_interface::JointStateHandle state_handle(joint_names_[i], &act_dis[i], &act_vel[i], &act_eff[i]);
+    hardware_interface::JointStateHandle state_handle(
+        joint_names_[i], &act_dis[i], &act_vel[i], &act_eff[i]);
     joint_state_interface.registerHandle(state_handle);
 
     // Connect and register the position_joint_interface
-    hardware_interface::JointHandle pos_handle(joint_state_interface.getHandle(joint_names_[i]), &cmd_dis[i]);
+    hardware_interface::JointHandle pos_handle(
+        joint_state_interface.getHandle(joint_names_[i]), &cmd_dis[i]);
     position_joint_interface.registerHandle(pos_handle);
+
+    // Connect and register the velocity_joint_interface
+    hardware_interface::JointHandle vel_handle(
+        joint_state_interface.getHandle(joint_names_[i]), &cmd_vel[i]);
+    velocity_joint_interface.registerHandle(vel_handle);
   }
   registerInterface(&joint_state_interface);
   registerInterface(&position_joint_interface);
+  registerInterface(&velocity_joint_interface);
 }
 
-FakeJointDriver::~FakeJointDriver()
-{
+FakeJointDriver::~FakeJointDriver() {}
+
+/**
+ * @brief Update function to call all of the update function of motors
+ */
+void FakeJointDriver::update(ros::Duration period) {
+
+  // create a lambda substract functor
+  std::function<double(double, double)> substractor = [](double a, double b) {
+    return std::abs(a - b);
+  };
+
+  // create a position difference vector
+  std::vector<double> pos_diff;
+  pos_diff.resize(cmd_dis.size());
+  std::transform(cmd_dis.begin(), cmd_dis.end(), cmd_dis_old.begin(),
+                 pos_diff.begin(), substractor);
```

Comment:
I think what you're doing here is pretty clever. Could be made faster with Eigen::VectorXd but this is fine

Reply:
This way we  avoid another dependency at least.